modApiServer/src/org/aion/api/server/rpc3/RPCMethodUtils.java

```diff
@@ -0,0 +1,378 @@
+                            .toArray(ByteArray[]::new));
+        }
+    }
+
+    static EthBlock serializeEthBlock(Block block, TransactionUnion txUnion) {
+        if (block.getHeader().getSealType().equals(BlockSealType.SEAL_POW_BLOCK)) {
+            AionBlock powBlock = ((AionBlock) block);
+            return new EthBlock(
+                    block.getNumber(),
+                    ByteArray.wrap(block.getHash()),
+                    ByteArray.wrap(block.getParentHash()),
+                    ByteArray.wrap(block.getLogBloom()),
+                    ByteArray.wrap(block.getTxTrieRoot()),
+                    ByteArray.wrap(block.getStateRoot()),
+                    ByteArray.wrap(block.getReceiptsRoot()),
+                    block.getDifficultyBI(),
+                    block.getTotalDifficulty(),
+                    block.getTimestamp(),
+                    block.getCoinbase(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    BlockSealType.SEAL_POW_BLOCK.getSealId(),
+                    block.isMainChain(),
+                    block.size(),
+                    txUnion,
+                    ByteArray.wrap(powBlock.getNonce()),
+                    ByteArray.wrap(powBlock.getHeader().getSolution()),
+                    null,
+                    null,
+                    null);
+        } else {
+            StakingBlock posBlock = ((StakingBlock) block);
+            return new EthBlock(
+                    block.getNumber(),
+                    ByteArray.wrap(block.getHash()),
+                    ByteArray.wrap(block.getParentHash()),
+                    ByteArray.wrap(block.getLogBloom()),
+                    ByteArray.wrap(block.getTxTrieRoot()),
+                    ByteArray.wrap(block.getStateRoot()),
+                    ByteArray.wrap(block.getReceiptsRoot()),
+                    block.getDifficultyBI(),
+                    block.getTotalDifficulty(),
+                    block.getTimestamp(),
+                    block.getCoinbase(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    block.getNrgConsumed(),
+                    block.getNrgLimit(),
+                    BlockSealType.SEAL_POW_BLOCK.getSealId(),
+                    block.isMainChain(),
+                    block.size(),
+                    txUnion,
+                    null,
+                    null,
+                    ByteArray.wrap(posBlock.getHeader().getSeed()),
+                    ByteArray.wrap(posBlock.getHeader().getSignature()),
+                    ByteArray.wrap(posBlock.getHeader().getSigningPublicKey()));
+        }
+    }
+
+    static EthTransactionForBlock serializeEthTransactionForBlock(
+            AionTransaction transaction, Block block, int transactionIndex) {
+        AionAddress contractAddress =
```

Comment:
The contractAddress is only meaningful when the transaction can be executed successful. Not sure the purpose to put the contract Address into the transaction. Maybe need to check the Transaction result and then put the address in.